include/rocksdb/db.h

```diff
@@ -969,6 +971,16 @@ class DB {
       const std::vector<ColumnFamilyHandle*>& column_families,
       std::vector<Iterator*>* iterators) = 0;
 
+  // DO NOT USE, UNDER CONSTRUCTION
+  // Return a cross-column-family iterator from a consistent database state.
+  // When the same key exists in more than one column families, the iterator
+  // selects the value from the first column family containing the key, in the
+  // order provided in the `column_families` parameter.
+  virtual MultiCfIterator* NewMultiCfIterator(
+      const ReadOptions& options,
+      const std::vector<ColumnFamilyHandle*>& column_families,
+      const Comparator* comparator = BytewiseComparator()) = 0;
```

Comment:
It feels dangerous to have a default comparator here. If we ask user to provide a comparator, we should probably specify in API comments that the input comparator should order the keys in a CF the same order as that CF's comparator. Maybe we can just use a given column families' comparator and not let user to specify one.

Reply:
There's actually a use case where cross-CF key comparator is different from within-CF key comparator. e.g. ZippyDB's metadata to be stored in a separate CF.  The key format may be different, but can be compared with a custom logic.

@ltamasi please let me know if my understanding is wrong here.

Reply:
... so in short, yeah, I think we can just error out during iterator creation if the comparators don't match.